lib/src/models/parsed_complex_native_script.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'parsed_native_script.dart';
+
+part 'parsed_complex_native_script.freezed.dart';
+
+@freezed
+class ParsedComplexNativeScript with _$ParsedComplexNativeScript {
```

Comment:
You should make this `sealed class` to fully benefit from the power of the dart 3 switch statements

Reply:
done in the other branch